GoogleForCreators/web-stories-wp
# Dashboard: Google Analytics id updates via settings view

## Summary
New flag to enableSettingsView that, when enabled, allows a user to update their Google Analytics Id if they have editing capabilities. 

## Relevant Technical Choices
- New flag called `enableSettingsView` to control settings view visibility. 
  - moved the settings page out from control of `inProgress` flagging 
- Sets up Karma for editorSettings view 
- Basic front end validation for the proper format of a google analytics id before sending input to API 
- Connects to settings API

## To Do (Amended)
_These should be follow up tickets/prs_ 

- Currently, the UI is not aware of if sitekit is installed or not per [product spec](https://docs.google.com/document/d/1Xs4uOZ3KCNlTMdjDT6q7cvTUpA1xk-kzrnVLQALltQk/edit?pli=1#)
- How to handle success of ID updates. Do we want a success toast? 

## User-facing changes
- When the flag is enabled *and you are logged in as an admin*, you should be able to navigate to settings from the dashboard and update the google analytics id.

## Testing Instructions
- Login as an Admin role and enable flag and see the above UI for validation. 
- Only input formatted as an id (ie: ua-000000-2) or empty input will call API 
- Only when the input is different from existing value will the API be called 
- you must hit 'enter' to save the ID update.
<img width="1019" alt="Screen Shot 2020-08-11 at 10 33 50 AM" src="https://user-images.githubusercontent.com/10720454/89929555-32c10080-dbbe-11ea-9b03-342b5364b13a.png">
<img width="1017" alt="Screen Shot 2020-08-11 at 10 33 41 AM" src="https://user-images.githubusercontent.com/10720454/89929563-33f22d80-dbbe-11ea-9ad6-24ca88b9c003.png">

- Login as any other role besides admin and see that settings is no longer available (this one is hard to test with the experiment flag UI since it doesn't show up - instead for this you can manually update `includes/Experiments.php` `enableSettingsView` to `'default' => true` (line 242). 


---



Addresses #3482
